core/arch/arm/kernel/thread_a32.S

```diff
@@ -40,9 +40,18 @@
 
 #include "thread_private.h"
 
+	.macro maybe_invalidate_bp
+#ifdef CFG_WITH_ARM_TRUSTED_FW
+	/* Invalidate the branch predictor. */
+	write_bpiallis
```

Comment:
Why is BP invalidate needed at all in this file when `CFG_WITH_ARM_TRUSTED_FW=y`? Isn't EL3 supposed to take care of it if needed? (I'm referring to ARM-TF pull request 1214).

Reply:
I was asked to do it like this, I guess this was before https://github.com/ARM-software/arm-trusted-firmware/pull/1214 existed.
@jbech-linaro, do you know if this is still needed?

Reply:
I think @jforissier is right here, we don't need to do it if it is done by Arm-TF. We need to make the branch invalidate when we are running our own monitor (and therefore responsible for the switch).

Reply:
OK, let's drop this.